convex/lib/withSession.ts

```diff
@@ -203,22 +215,26 @@ export function queryWithSession<
   Output
 >(
   func: ValidatedFunction<
-    QueryCtx & { session: Doc<"sessions"> | null },
+    QueryCtx & { session: Doc<"sessions"> },
     ArgsValidator,
     Promise<Output>
   >
 ): RegisteredQuery<
   "public",
-  [ObjectType<ArgsValidator> & { sessionId: Id<"sessions"> | null }],
+  ObjectType<ArgsValidator> & ObjectType<typeof sessionMiddlewareValidator>,
   Output
 >;
-export function queryWithSession<Output>(
+export function queryWithSession<Args extends ArgsArray, Output>(
   func: UnvalidatedFunction<
-    QueryCtx & { session: Doc<"sessions"> | null },
-    [],
+    QueryCtx & { session: Doc<"sessions"> },
+    Args,
     Promise<Output>
   >
-): RegisteredQuery<"public", [{ sessionId: Id<"sessions"> | null }], Output>;
+): RegisteredQuery<
+  "public",
+  MergeArgsForRegistered<Args, ObjectType<typeof sessionMiddlewareValidator>>,
+  Output
+>;
 export function queryWithSession(func: any): any {
-  return query(withOptionalSession(func));
+  return query(withSession(func));
```

Comment:
query is no longer optional by default? I found, without adding an app-wide spinner for sessions, that session queries were the ones most likely to need to handle un-initialized sessions